src/validation.js

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+// ---------------------------------------------------------------------
+
+/* eslint-disable complexity, max-depth, no-extra-parens, no-magic-numbers */
+
+export class AuroFormValidation {
+
+  /**
+   * Generates a camelCase version of the tag name for an element.
+   * @private
+   * @param {object} elem - HTML element with tagname to convert.
+   * @returns {string} Tag name in camelCase syntax.
+   */
+  getCamelCaseName(elem) {
+    const tagName = elem.tagName.toLowerCase();
+    const tagNameDivider = tagName.indexOf('-');
+    const nameSpace = tagName.substring(0, tagNameDivider);
+    const name = tagName.slice(tagNameDivider + 1);
+    const camelCaseName = nameSpace + name.charAt(0).toUpperCase() + name.slice(1);
+
+    return camelCaseName;
+  }
+
+  /**
+   * Determines the validity state of the element based on the common attribute restrictions (pattern).
+   * @private
+   * @param {object} elem - HTML element to validate.
+   * @returns {void}
+   */
+  validateAttributes(elem) {
+    if (elem.pattern) {
+      const pattern = new RegExp(`^${elem.pattern}$`, 'u');
+
+      if (!pattern.test(elem.value)) {
+        elem.validity = 'badInput';
+        elem.setCustomValidity = elem.setCustomValidityBadInput || '';
+      }
+    } else if (elem.value.length > 0 && elem.value.length < elem.minLength) {
+      elem.validity = 'tooShort';
+      elem.setCustomValidity = elem.setCustomValidityTooShort || '';
+    } else if (elem.value.length > elem.maxLength) {
+      elem.validity = 'tooLong';
+      elem.setCustomValidity = elem.setCustomValidityTooLong || '';
+    }
+  }
+
+  /**
+   * Determines the validity state of the element based on the type attribute.
+   * @private
+   * @param {object} elem - HTML element to validate.
+   * @returns {void}
+   */
+  validateType(elem) {
+    if (elem.hasAttribute('type')) {
+      if (elem.type === 'email') {
+        // BUG - Need more accurate email regex | this one validates slightly different than the default html5 one
```

Comment:
What's going on here?

Reply:
This is an old comment. The regex we have works but it is worth considering if there is a better way to write this regex. There are several approaches out there.

Reply:
Please address this comment

Reply:
I opened a separate ticket to track this: https://github.com/AlaskaAirlines/auro-formvalidation/issues/3